Second instrumentation module breaks whole maven build

We are using Maven and we have big project that contains 2 applications and common libraries. Recently I decided to write instrumentation tests for both application and launch them with Spoon. At second instrumentation test whole build failed

```[ERROR] Failed to execute goal com.squareup.spoon:spoon-maven-plugin:1.0.3:run (default) on project caregiver-it: Execution default of goal com.squareup.spoon:spoon-maven-plugin:1.0.3:run failed. NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.squareup.spoon:spoon-maven-plugin:1.0.3:run (default) on project caregiver-it: Execution default of goal com.squareup.spoon:spoon-maven-plugin:1.0.3:run failed.
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:225)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:319)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:156)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:537)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:196)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:141)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:601)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:290)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:230)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:409)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:352)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default of goal com.squareup.spoon:spoon-maven-plugin:1.0.3:run failed.
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:110)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:209)
	... 19 more
Caused by: java.lang.NullPointerException
	at com.android.ddmlib.AndroidDebugBridge.stop(AndroidDebugBridge.java:691)
	at com.android.ddmlib.AndroidDebugBridge.createBridge(AndroidDebugBridge.java:296)
	at com.squareup.spoon.SpoonUtils.initAdb(SpoonUtils.java:121)
	at com.squareup.spoon.SpoonRunner.run(SpoonRunner.java:68)
	at com.squareup.spoon.SpoonMojo.execute(SpoonMojo.java:154)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:101)
	... 20 more```

Separately those tests work fine

--- a/spoon-runner/src/main/java/com/squareup/spoon/SpoonRunner.java
+++ b/spoon-runner/src/main/java/com/squareup/spoon/SpoonRunner.java
@@ -114,6 +114,7 @@ public final class SpoonRunner {
 
       return parseOverallSuccess(summary);
     } finally {
+      AndroidDebugBridge.disconnectBridge();
       AndroidDebugBridge.terminate();
     }
   }
@@ -162,8 +163,7 @@ public final class SpoonRunner {
         final String safeSerial = SpoonUtils.sanitizeSerial(serial);
         logDebug(debug, "[%s] Starting execution.", serial);
         Runnable runnable = new Runnable() {
-          @Override
-          public void run() {
+          @Override public void run() {
             try {
               summary.addResult(safeSerial, getTestRunner(serial, testInfo).runInNewProcess());
             } catch (Exception e) {
@@ -379,22 +379,22 @@ public final class SpoonRunner {
   }
 
   static class CommandLineArgs {
-    @Parameter(names = { "--title" }, description = "Execution title")
+    @Parameter(names = { "--title" }, description = "Execution title") //
     public String title = DEFAULT_TITLE;
 
     @Parameter(names = { "--apk" }, description = "Application APK",
-        converter = FileConverter.class, required = true)
+        converter = FileConverter.class, required = true) //
     public File apk;
 
     @Parameter(names = { "--test-apk" }, description = "Test application APK",
-        converter = FileConverter.class, required = true)
+        converter = FileConverter.class, required = true) //
     public File testApk;
 
     @Parameter(names = { "--class-name" }, description = "Test class name to run (fully-qualified)")
     public String className;
 
     @Parameter(names = { "--method-name" },
-        description = "Test method name to run (must also use --class-name)")
+        description = "Test method name to run (must also use --class-name)") //
     public String methodName;
 
     @Parameter(names = { "--size" }, converter = TestSizeConverter.class,
@@ -402,31 +402,31 @@ public final class SpoonRunner {
     public IRemoteAndroidTestRunner.TestSize size;
 
     @Parameter(names = { "--output" }, description = "Output path",
-        converter = FileConverter.class)
+        converter = FileConverter.class) //
     public File output = cleanFile(SpoonRunner.DEFAULT_OUTPUT_DIRECTORY);
 
-    @Parameter(names = { "--sdk" }, description = "Path to Android SDK")
+    @Parameter(names = { "--sdk" }, description = "Path to Android SDK") //
     public File sdk = cleanFile(System.getenv("ANDROID_HOME"));
 
     @Parameter(names = { "--fail-on-failure" }, description = "Non-zero exit code on failure")
     public boolean failOnFailure;
 
     @Parameter(names = { "--fail-if-no-device-connected" },
-        description = "Fail if no device is connected")
+        description = "Fail if no device is connected") //
     public boolean failIfNoDeviceConnected;
 
     @Parameter(names = { "--sequential" },
-        description = "Execute tests sequentially (one device at a time)")
+        description = "Execute tests sequentially (one device at a time)") //
     public boolean sequential;
 
     @Parameter(names = { "--no-animations" }, description = "Disable animated gif generation")
     public boolean noAnimations;
 
     @Parameter(names = { "--adb-timeout" },
-        description = "Set maximum execution time per test in seconds (10min default)")
+        description = "Set maximum execution time per test in seconds (10min default)") //
     public int adbTimeoutSeconds = DEFAULT_ADB_TIMEOUT;
 
-    @Parameter(names = { "--debug" }, hidden = true)
+    @Parameter(names = { "--debug" }, hidden = true) //
     public boolean debug;
 
     @Parameter(names = { "-h", "--help" }, description = "Command help", help = true, hidden = true)

